tests/test_nodes.py

```diff
@@ -660,6 +661,26 @@ def test_decay_queue(self):
         }
         self.assertDictAlmostEqual(d4, tank.storage, 15)
 
+    def test_node_overrides(self):
+        node = Node(name="", data_input_dict={("temperature", 1): 15})
+        # check the format of dict
+        new_data_input_dict = {
+            ("temperature", 1): 10,
+            ("temperature", 2): 20,
+        }
+        node.apply_overrides({"data_input_dict": new_data_input_dict})
+        self.assertDictEqual(node.data_input_dict, new_data_input_dict)
+        # check the format of str
+        new_data_input_dict = str(
+            Path(__file__).parent / "example_data_input_dict.csv.gz"
+        )
+        node.apply_overrides({"data_input_dict": new_data_input_dict})
+        from wsimod.orchestration.model import read_csv
```

Comment:
You need some consistency with the imports. Either import everything at the top of the file or at the top of the function, but not in the middle of it - unless there is a very good reason, like using patching and things like that, which is not the case.

Reply:
I'll wait the conversation on `read_csv` to be resolved to address this.

Reply:
@dalonsoa If I move this `from wsimod.orchestration.model import read_csv` outsidethe `apply_overrides` function to the beginning, I will have the following error which breaks 11 `test` scripts.

```
Traceback (most recent call last):

  File "C:\Users\ll6715\Anaconda3\envs\wsimod\lib\site-packages\spyder_kernels\py3compat.py", line 356, in compat_exec
    exec(code, globals, locals)

  File "c:\users\ll6715\documents\github\wsi\tests\test_nodes.py", line 14, in <module>
    from wsimod.nodes.nodes import Node

  File "c:\users\ll6715\documents\github\wsi\wsimod\nodes\__init__.py", line 2, in <module>
    from wsimod.nodes.catchment import Catchment

  File "c:\users\ll6715\documents\github\wsi\wsimod\nodes\catchment.py", line 9, in <module>
    from wsimod.nodes.nodes import Node

  File "c:\users\ll6715\documents\github\wsi\wsimod\nodes\nodes.py", line 13, in <module>
    from wsimod.orchestration.model import read_csv

  File "c:\users\ll6715\documents\github\wsi\wsimod\orchestration\model.py", line 21, in <module>
    from wsimod.nodes.land import ImperviousSurface

  File "c:\users\ll6715\documents\github\wsi\wsimod\nodes\land.py", line 12, in <module>
    from wsimod.nodes.nodes import Node

ImportError: cannot import name 'Node' from partially initialized module 'wsimod.nodes.nodes' (most likely due to a circular import) (c:\users\ll6715\documents\github\wsi\wsimod\nodes\nodes.py)
```

But if it stays in the function, no errors reported from the tests.

Reply:
So don't move it. In this particular case, it needs to be where it is, otherwise, there are circular imports. The general rule is "at the beginning", but there are reasons for exceptions.